engine/tester/paymailmock/http_sniffer.go

```diff
@@ -0,0 +1,79 @@
+package paymailmock
+
+import (
+	"bytes"
+	"io"
+	"net/http"
+	"regexp"
+
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+)
+
+// CallDetails holds the details of a call made to the mocked server
+type CallDetails struct {
+	URL string
+
+	RequestMethod string
+	RequestBody   []byte
+
+	ResponseBody []byte
+	ResponseCode int
+}
+
+type httpSniffer struct {
+	next   http.RoundTripper
+	called map[string]CallDetails
```

Comment:
MINOR 🤔 

Maybe it's a bit redundant here (as it's made for testing) - but shouldn't it be a safe-thread map? 

If not, maybe add some comment that this httpSniffer is designed to be used in sequential tests only or be a fresh instance with every test case?

I am a bit worried that in two separated goroutines this RoundTrip function my write details to a map simultaneously.
Or maybe it's not a problem at all 🤷